Detectors/TPC/qc/src/DCSPTemperature.cxx

```diff
@@ -0,0 +1,165 @@
+
+  auto cStats = new TCanvas("c_temperature_stats", "c_temperature_stats", 1000, 1000);
+  auto cSensorsMultiGraph = new TCanvas("c_temperature_sensors", "c_temperature_sensors", 1000, 1000);
+  // auto cSensorsMean = new TCanvas("c_temperature_sensors_single", "c_temperature_sensors_single", 1000, 1000);
+
+  mCanVec.emplace_back(cStats);
+  mCanVec.emplace_back(cSensorsMultiGraph);
+  // mCanVec.emplace_back(cSensorsMean);
+
+  gStyle->SetTimeOffset(0);
+}
+
+void DCSPTemperature::processData(const std::vector<o2::tpc::dcs::Temperature*>& data)
+{
+  int nPointA = 0;
+  int nPointC = 0;
+  int sensorCounter = 0;
+
+  auto grM = new TMultiGraph;
+  std::vector<TGraph*> grVec; // one graph per sensor
+  auto grStatsAMean = new TGraph;
+  auto grStatsAGradX = new TGraph;
+  auto grStatsAGradY = new TGraph;
+  auto grStatsCMean = new TGraph;
+  auto grStatsCGradX = new TGraph;
+  auto grStatsCGradY = new TGraph;
+
+  std::vector<int> pointCountersSensors;
+
+  for (int i = 0; i < 18; i++) {
+    pointCountersSensors.emplace_back(0);
+    grVec.emplace_back(new TGraph);
+    grVec.back()->SetNameTitle(fmt::format("tempSensor{}", i + 1).data(), fmt::format("Temperature Sensor {};;T (K)", i + 1).data());
+    grVec.back()->GetXaxis()->SetTimeDisplay(1);
+    grVec.back()->GetXaxis()->SetTimeFormat("#splitline{%d.%m.%y}{%H:%M:%S}");
+    grVec.back()->SetMarkerStyle(20);
+  }
+
+  for (const auto& t : data) {
+    sensorCounter = 0;
+    for (const auto sensor : t->raw) {
+      for (const auto& value : sensor.data) {
+        grVec.at(sensorCounter)->SetPoint(pointCountersSensors.at(sensorCounter), double(value.time) / 1000., value.value);
+        pointCountersSensors.at(sensorCounter)++;
+      }
+      sensorCounter++;
+    }
+    for (const auto& value : t->statsA.data) {
+      grStatsAMean->SetPoint(nPointA, double(value.time) / 1000., value.value.mean);
+      grStatsAGradX->SetPoint(nPointA, double(value.time) / 1000., value.value.gradX);
+      grStatsAGradY->SetPoint(nPointA, double(value.time) / 1000., value.value.gradY);
+      nPointA++;
+    }
+    for (const auto& value : t->statsC.data) {
+      grStatsCMean->SetPoint(nPointC, double(value.time) / 1000., value.value.mean);
+      grStatsCGradX->SetPoint(nPointC, double(value.time) / 1000., value.value.gradX);
+      grStatsCGradY->SetPoint(nPointC, double(value.time) / 1000., value.value.gradY);
+      nPointC++;
+    }
+  }
+
+  for (auto& gr : grVec) {
+    grM->Add(gr);
+  }
+
+  // mCanVec.at(2)->DivideSquare(grVec.size());
+
+  /*int padIter = 1;
+  for (auto& gr : grVec) {
+    mCanVec.at(2)->cd(padIter);
+    gr->Draw("apl");
+    padIter++;
+  }*/
+
+  mCanVec.at(0)->Divide(2, 3);
+
+  grStatsAMean->SetTitle("Temperature mean (A-side);;T (K)");
+  grStatsAMean->SetMarkerStyle(20);
+  grStatsAMean->GetXaxis()->SetTimeDisplay(1);
+  grStatsAMean->GetXaxis()->SetTimeFormat("#splitline{%d.%m.%y}{%H:%M:%S}");
+  mCanVec.at(0)->cd(1);
+  grStatsAMean->Draw("apl");
```

Comment:
```suggestion
grStatsAMean->Sort();
  grStatsAMean->Draw("apl");
```
`Sort` also all other graphs. Looking at qcg it seems that the data points are not necessarily ordered in time...

Reply:
I think what you mean in the qcg is the result of the missing `clear` in the uploaded data. I can of course sort the graphs anyway since it won't hurt.

Reply:
This is because of the missing `clear` in the uploaded data but I sort the data anyway since it doesn't hurt.